File: st-exerciseviewer/src/main/java/de/saring/exerciseviewer/gui/EVDocument.java
package de.saring.exerciseviewer.gui;

import de.saring.exerciseviewer.core.EVException;
import de.saring.exerciseviewer.core.EVOptions;
import de.saring.exerciseviewer.data.EVExercise;
import de.saring.exerciseviewer.parser.ExerciseParser;
import de.saring.exerciseviewer.parser.ExerciseParserFactory;

/**
 * This class contains all document (MVC) related data and functionality of the
 * ExerciseViewer application.
 *
 * @author Stefan Saring
 * @version 1.0
 */
public class EVDocument {

    /**
     * The current exercise to be displayed.
     */
    private EVExercise exercise;

    /**
     * The filename of the current exercise.
     */
    private String exerciseFilename;

    /**
     * The ExerciseViewer options.
     */
    private EVOptions options;


    /**
     * Reads the specified exercise file and stores it in the document.
     *
     * @param filename exercise filename
     * @throws EVException on parsing problems
     */
    public void openExerciseFile(String filename) throws EVException {

        ExerciseParser parser = ExerciseParserFactory.getParser(filename);
        exercise = parser.parseExercise(filename);
        exerciseFilename = filename;
    }

    public EVExercise getExercise() {
        return exercise;
    }

    public String getExerciseFilename() {
        return exerciseFilename;
    }

    public EVOptions getOptions() {
        return options;
    }

    public void setOptions(EVOptions options) {
        this.options = options;
    }
}

File: st-exerciseviewer/src/main/kotlin/de/saring/exerciseviewer/gui/EVDocument.kt
package de.saring.exerciseviewer.gui

import de.saring.exerciseviewer.core.EVException
import de.saring.exerciseviewer.core.EVOptions
import de.saring.exerciseviewer.data.EVExercise
import de.saring.exerciseviewer.parser.ExerciseParserFactory
import de.saring.util.unitcalc.FormatUtils.SpeedMode

/**
 * This class contains all model / document (MVC) related data and functionality of the ExerciseViewer application.
 *
 * @property options The ExerciseViewer options.
 *
 * @author Stefan Saring
 */
class EVDocument(val options: EVOptions) {

    /** The current exercise to be displayed.  */
    lateinit var exercise: EVExercise

    /** The filename of the current exercise.  */
    lateinit var exerciseFilename: String

    /** The speed mode to be used for showing the speed values of the current exercise. */
    lateinit var speedMode: SpeedMode

    /**
     * Reads the specified exercise file and stores it in the document.
     *
     * @param filename exercise filename
     * @param speedMode the speed mode to be used for showing speed values
     * @throws EVException on parsing problems
     */
    fun openExerciseFile(filename: String, speedMode: SpeedMode) {

        val parser = ExerciseParserFactory.getParser(filename)
        exercise = parser.parseExercise(filename)
        exerciseFilename = filename
        this.speedMode = speedMode
    }
}
